File: src/main/java/org/goafabric/personservice/logic/PersonMapper.java
package org.goafabric.personservice.logic;

import org.goafabric.personservice.controller.dto.Address;
import org.goafabric.personservice.controller.dto.Person;
import org.goafabric.personservice.controller.dto.PersonSearch;
import org.goafabric.personservice.persistence.entity.AddressEo;
import org.goafabric.personservice.persistence.entity.PersonEo;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

import java.util.List;


@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface PersonMapper {
    Person map(PersonEo value);

    PersonEo map(Person value);

    List<Person> map(List<PersonEo> value);

    List<Person> map(Iterable<PersonEo> value);

    PersonEo map(PersonSearch value);

    Address map(AddressEo value);
}

File: src/main/java/org/goafabric/personservice/logic/PersonMapper.kt
package org.goafabric.personservice.logic

import org.goafabric.personservice.persistence.domain.PersonBo
import org.goafabric.personservice.controller.dto.Person
import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
interface PersonMapper {
    fun map(person: PersonBo): Person
    fun map(person: Person): PersonBo
    fun map(countries: List<PersonBo>): List<Person>
}